docs/source/final_attrs.rst

```diff
@@ -0,0 +1,267 @@
+
+* A final attribute can't be overridden by a subclass (even with another
+  explicit final declaration). Note however, that final attributes can
+  override read-only properties. This also applies to multiple inheritance:
+
+  .. code-block:: python
+
+     class Base:
+         @property
+         def ID(self) -> int: ...
+
+     class One(Base):
+         ID: Final = 1  # OK
+
+     class Other(Base):
+         ID: Final = 2  # OK
+
+     class Combo(One, Other):  # Error: cannot override final attribute.
+         pass
+
+* Declaring a name as final only guarantees that the name wll not be re-bound
+  to other value, it doesn't make the value immutable. One can use immutable ABCs
+  and containers to prevent mutating such values:
+
+  .. code-block:: python
+
+     x: Final = ['a', 'b']
+     x.append('c')  # OK
+
+     y: Final[Sequance[str]] = ['a', 'b']
+     y.append('x')  # Error: Sequance is immutable
+     z: Final = ('a', 'b')  # Also an option
+
+Final methods
+*************
+
+Like with attributes, sometimes it is useful to protect a method from
+overriding. In such situations one can use the ``typing_extensions.final``
+decorator:
+
+.. code-block:: python
+
+   from typing_extensions import final
+
+   class Base:
+       @final
+       def common_name(self) -> None:
+           ...
+
+   # 1000 lines later
+
+   class Derived(Base):
+       def common_name(self) -> None:  # Error: cannot override a final method
+           ...
+
+This ``@final`` decorator can be used with instance methods, class methods,
+static methods, and properties (this includes overloaded methods). For
+overloaded methods one should add ``@final`` on the implementation to make
+it final (or on the first overload in stubs):
+
+.. code-block:: python
+   from typing import Any, overload
+
+   class Base:
+       @overload
+       def meth(self) -> None: ...
+       @overload
+       def meth(self, arg: int) -> int: ...
+       @final
+       def meth(self, x=None):
+           ...
+
+Final classes
+*************
+
+You can apply a ``typing_extensions.final`` decorator to a class to indicate
+to mypy that it can't be subclassed. The decorator acts as a declaration
+for mypy (and as documentation for humans), but it doesn't prevent subclassing
+at runtime:
+
+.. code-block:: python
+
+   from typing_extensions import final
+
+   @final
+   class Leaf:
+       ...
+
+   class MyLeaf(Leaf):  # Error: Leaf can't be subclassed
+       ...
+
+Here are some situations where using a final class may be useful:
+
+* A class wasn't designed to be subclassed. Perhaps subclassing does not
+  work as expected, or it's error-prone.
+* You want to retain the freedom to arbitrarily change the class implementation
+  in the future, and these changes might break subclasses.
+* You believe that subclassing would make code harder to understand or maintain.
+  For example, you may want to prevent unnecessarily tight coupling between
+  base classes and subclasses.
```

Comment:
These bullets all make almost the same point. Maybe we don't need to editorialize and users can figure out for themselves why final classes might be useful (they might even know the concept from another language).

Reply:
IIRC, Jukka proposed to add some examples, and I think it probably makes sense to keep at least some of them.